### DISCUSSION TITLE
Default setting/bindings changes for v2.0.0

### DISCUSSION THREAD

**Discussion by The-Compiler** [qutebrowser/qutebrowser#5999] [General] [answered]
For the v2.0.0 release, I'd like to clean up some settings and perhaps also change some default values or bindings. Let's use this discussion to gather and vote/discuss about various proposals in that direction.

To keep things clean, please follow a couple of guidelines:

- If you have a new suggestion about a default binding/setting to be changed, add a new post with the text input at the very bottom of the page, so that every suggestion starts a new thread.
- If you have a comment on one of the suggestions, do **not** post it as a new post at the very bottom, instead reply directly to the suggestion in question by using the text input next to it.
- To vote on suggestions, use the :+1: and  :-1: reactions. Don't use comments for a simple "good idea" reply, but please *do* add comments if you want to elaborate on why you voted the way you did, especially for :-1: votes.
- Note that the changes should be rather small/incremental, and ideally uncontroversial. The original philosophy behind the keyboard bindings was to be in-line with what [dwb](https://wiki.archlinux.org/index.php/DWB) did, and many qutebrowser users today still are people coming from dwb. Changing all of the bindings to be e.g. equal to the one Tridactyl/Vimperator/Vimium/... uses is out of the question.
- To post a new proposal, ideally use this template to make things easier to scan:

```md
## Title

Text
```

**Comment by The-Compiler** [comment]
## Setting tabs.background by default

Right now, `tabs.background` is set to `false` by default - it's something that always trips me up when e.g. running with `--temp-basedir`, and I think it's also the first setting people set to `true`. It seems to me like it should be `true` by default?

**Reply by brianbuccola** [reply]
Really? I've always kept mine as `false`. If I want to open multiple background tabs, then I use rapid hints (`;r`), rather than middle-clicking a bunch of links.

**Comment by The-Compiler** [comment]
## Rebinding d to dd

This came up a couple of times recently - some people would like to see `dd` rather than `d` to close tabs, for various reasons:

- It's closer to how `dd` works in vim
- While `u` (`:undo`) is available, some people might not know about it
- Even if you know about `u`, some state (filled out forms and such) might get lost when closing/reopening tabs, so it's somewhat of a destructive action

I suppose it'd also give us some more `d<something>` bindings, though I can't think of anything right now.

Personally, I'm not sure I like the idea because pretty much all other vim-like browsers (I think) use `d` (or another single key) to close tabs, but I'm open to changing it.

**Reply by rien333** [reply]
> I suppose it'd also give us some more d<something> bindings, though I can't think of anything right now.

I have bindings for "close all" (could be bound to <kbd>da</kbd>, perhaps? though it's <kbd>co</kbd> for me). I also have bindings for "close all tabs to the left" (<kbd>dl</kbd>), and "close all to the right" (<kbd>dr</kbd>). (I have those prefixed to "c" though)

Still, many single keyboard driven interfaces uses a solitary <kbd>d</kbd> to delete things (e.g. file managers, like the emacs one). Hence, I don't think a sole <kbd>d</kbd> is that confusing (espcially because if you happen to come in with expectancy that <kbd>dd</kbd> closes tabs, you'll quickly discover that one keypress is enough)

**Reply by nichobi** [reply]
I have this binding mainly because `hints.leave_on_load` is enabled by default, and since `d` is a default hint key I often ended up closing tabs unexpectedly, as pages were still loading in. I can imagine many other beginners run into the same issue

**Reply by The-Compiler** [reply]
> I have this binding mainly because `hints.leave_on_load` is enabled by default

Which might be another potential change - I just [added a proposal](https://github.com/qutebrowser/qutebrowser/discussions/5999#discussioncomment-254476) for that one as well.

**Reply by gavtroy** [reply]
The real issue is how destructive accidentally closing a tab is. Ideally there'd be a grace period allowing a complete and instant undo, during which the tab is hidden but not actually destroyed. This might be a better long-term direction than making the bindings more awkward.

**Reply by toofar** [reply]
@gavtroy Sometime I close tabs to mitigate a lack of resources, for example the machine is starting to swap. Depending on whether there was anything merged that uses the tab freezing thing (and how that deals with low memory situations) a grace period wouldn't be very helpful in that situation. Although a force-close might help.

**Reply by gerion0** [reply]
I'm hoping that tree tabs will be part of qutebrowser some day. When that is the case, it is meaningful to have also a binding for "close whole tab group", so I'm in favor for `dd`.

**Reply by The-Compiler** [reply]
I opened https://github.com/qutebrowser/qutebrowser/issues/6064 for this.

**Reply by Phantop** [reply]
I was about to suggest adding a force close, as toofar did, and binding it to `D` if we add in the grace period, but it's already used by the `tab-close -o` binding I wasn't even aware of. Don't know how many others use that but if that one is done there should be a force close binding of some sort.

Alternatively, go for `dd` and make `D` do the old behavior because one could instead do `dj` or `dk`  to emulate the `-o` option and `dK` and `dJ` would close the prior and next tab respecitvely.

**Comment by The-Compiler** [comment]
## Setting tabs.last_close to close by default

Some people seem to be confused about `d` doing nothing with a single tab. Now that we have window-undo (`U`), perhaps the default should be to close the window rather than doing nothing?

**Reply by rcorre** [reply]
I use `blank`. That avoids the confusion that it "did nothing", but avoids unintentionally closing the whole browser (though the latter isn't a huge deal if you've got no tabs open, so I think `close` is fine too).

**Comment by The-Compiler** [comment]
## Renaming qutebrowser userscripts to "qutescripts"

Not really a config change but still something I'd like some feedback on:

qutebrowser's [userscripts](https://github.com/qutebrowser/qutebrowser/blob/master/doc/userscripts.asciidoc) (a term inherited from [dwb](https://wiki.archlinux.org/index.php/DWB#Userscripts)) often get mixed up with [Greasemonkey userscripts](https://en.wikipedia.org/wiki/Userscript), despite being a totally different concept.

I feel like we should find a new name to (hopefully) reduce confusion. What about "qutescripts"? Or maybe someone has another nice proposal I didn't think of yet?

**Reply by Foxboron** [reply]
`qbscripts` pronounced `qubescripts`? Might be non obvious but shorter.

**Reply by drlobes** [reply]
If we aim for shorter, just qscripts sounds good as well.

**Reply by The-Compiler** [reply]
I don't think it needs to be short necessarily - it's not like the length of "userscripts" has ever bothered anyone (I'm assuming). I'd rather have something which is easy to grasp, ideally even if you hear it without explicitly saying "qutebrowser qscripts" or whatever.

**Reply by rien333** [reply]
In that case, I really like qutescripts.

**Reply by The-Compiler** [reply]
I opened https://github.com/qutebrowser/qutebrowser/issues/6063 for this.

**Comment by The-Compiler** [comment]
## Setting hints.leave_on_load to false by default

Given that we leave modes when loading *finished* rather than started (see #5931), doing so can be rather disruptive.

For `input.insert_mode.auto_leave`, I think that's worth the pain (until we have a better solution, at least) - staying in insert mode after e.g. starting a search is definitely rather confusing.

For `hints.leave_on_load` I'm not so sure. If a hint gets selected, hint mode is left anyways. The only case it handles, unless I'm missing something, is either the page loading something new; or the user clicking on something while in hint mode. That seems like a bit of a corner case, and probably not worth the pain caused by leaving the hint mode right when someone wants to follow a hint.

Thus, maybe it should be false by default?

**Comment by The-Compiler** [comment]
## Swapping J and K

Another thing that comes up in a while, see point 15 in the [FAQ](https://github.com/qutebrowser/qutebrowser/blob/master/doc/faq.asciidoc).

I lean towards not changing that, mainly because it's quite disruptive to everyone who's been using qutebrowser with the default keybindings (and because it makes no sense with vertical tabs) - but I'm curious what others think.

**Reply by drlobes** [reply]
I'm a new qb user and never used vim, I kid you not this is the first setting I did to qb, swap J and K because thats how it feels more natural to me, but thats something I can easily do again and not a reason to change it because I've seen so many using the default way.

**Reply by brianbuccola** [reply]
Switching tabs is so common that I never quite understood why vim-like browsers use modifier+key combinations for this. I prefer single keys, and so I just use `h` and `l`, which has the additional advantage that (for horizontal tabs) it corresponds to moving leftward and rightward, respectively. (To scroll leftward/rightward, which is much less common for me, I use `<C-h>` and `<C-l>`.)

**Reply by palb91** [reply]
By default, tabs are horizontal and when you run `:history`, the output is vertical (or history suggestion when running `:open`).
I found quite weird that <kbd>J</kbd><kbd>K</kbd> stand for next/prev-tab (horizontal) and <kbd>H</kbd><kbd>L</kbd> stands for prev/next history (so vertical). Even if I personnaly remapped all of them, may I suggest to invert them?

**Comment by The-Compiler** [comment]
## Setting content.cookies.accept to no-3rdparty

This actually was the default until v1.5.2, where it was changed to `all` because `no-3rdparty` broke various pages such as GMail.

If this isn't the case anymore, we should probably move back to `no-3rdparty`, given that there's a general movement towards blocking third-party cookies in other browsers as well.

**Comment by goosewood** [comment]
Instead of :buffer being accessed through gt, have it optionally also appear in the completion menu. 

I personally do forget what tabs I have open, especially with multiple windows and whatnot, and often find myself having multiple hope pages of the same site open. Personally I think it would help me save on ram, as I don't actively remember to check my buffer menu for an item frequently, particularly with site I frequent and have aliases for.

**Reply by milomindrbindr** [reply]
That would be awesome!

**Reply by The-Compiler** [reply]
I think this is more of a general feature request and rather out of scope for "simple" default configuration changes. 

I'm assuming with "the completion menu", you mean the `:open` completion? See #3194.

**Comment by The-Compiler** [comment]
## Rename `:buffer` to `:tab-select`

In qutebrowser, we don't really call tabs "buffers" (among other reasons, to be more accessible to people who aren't necessarily used to vim) - however, `:buffer` is kind of an oddball there. I'd like to have a `:tab-*` name for it as well (to also group it with other tab commands). What about `:tab-select`?

**Reply by arza-zara** [reply]
What about merging `:buffer` with `:tab-focus`? With flag `--best-match` or something?

**Reply by The-Compiler** [reply]
@arza-zara I'd rather not - `:tab-focus` already does many different things:

- It behaves like `:tab-next` with no count so you can use `T` to go to a given tab; or the next one
- But it can also be used as `:tab-focus last` to jump between tabs
- Or take an index as argument rather than count (so you can bind things to `:tab-focus 1` rather than `:run-with-count 1 tab-focus`)

While I like combining related commands, I feel like adding another way of using it seems like too much crammed into one.

**Comment by The-Compiler** [comment]
## Rename `:set` to `:config-set`

All config commands do have a `:config-` prefix, except for `:set` - for historical reasons, for consistency with vim, and because it's probably used more than the others. I wonder if it should perhaps be renamed to `:config-set` to be consistent with the rest instead?

**Reply by wishfort36** [reply]
I don't think that `:set` being renamed is the way to go. `:set` is one of the config-related commands I myself use the most, but removing `:set` won't just remove consistency with vim. It will remove consistency a heap of "vim-like" programs. To name a few I'm familiar with: ranger, zathura, mutt, cmus, newsboat and tmux. To those in "the ecosystem", `:set` would be the most intuitive way of configuring a program. It might be a better idea to make `:set` an abbreviation of `:config-set` in the same way that vim has abbreviations, such as `:s` being an abbreviation of `:substitute`.

**Comment by The-Compiler** [comment]
## Rename :open-editor to :edit-text

For consistency with `:edit-url` and `:edit-command`

**Reply by ghost** [reply]
This is a good idea because it still keeps "`edit`" in there for completion purposes. +1

**Comment by The-Compiler** [comment]
## Rename :record-macro and :run-macro

To `:macro-record` and `:macro-run` for consistency with other commands (and to group them, since they are related)

**Comment by The-Compiler** [comment]
## Rename :*-selection

Probably not much of a visible change, since those are used via keybindings only typically.

For consistency with other commands, `:toggle-selection`, `:drop-selection` and `:reverse-selection` should have the words swapped. Perhaps rename `:follow-selected` to `:selection-follow` as well. For consistency, it'd make sense to rename `:follow-hint` to `:hint-follow` too.

**Comment by gerion0** [comment]
Just got an idea when reading this. The *Setting content.cookies.accept to no-3rdparty* question is very much privacy related. I have also in my config:
 ```
content.cookies.store:
    global: false
content.javascript.enabled:
    global: false
content.autoplay:
    global: false
```
These settings are also privacy related but clearly too restrictive as a default setting (maybe the autoplay option is discussable). What do you think of a meta option/action: `:enable-privacy-settings` which sets exactly these kind of options, so the user doesn't need to fiddle them out by herself?

**Reply by rcorre** [reply]
Not a bad idea to make that easier, but you have to _know_ the special command exists. If we're going to have a doc saying "run :enable-privacy-settings", we might as well just say "set these options" in the doc instead. A command is slightly more discoverable (just by stumbling across it in command completion), but I don't think discoverability justifies a whole new command.

**Reply by gerion0** [reply]
Hmm, you're right. The only advantage I could think of is that a command will be probably better tested and updated, simply because it is part of the code. Anyway, a doc section would serve well for me.

**Reply by The-Compiler** [reply]
Privacy vs. working functionality isn't a black-and-white thing. I agree this is something that should be documented better (also see #4045; though #345 will need to be completed before I will work on writing more documentation).

Even now, there are many people who blindly e.g. do [things mentioned in the Arch Wiki](https://wiki.archlinux.org/index.php/qutebrowser#Minimize_fingerprinting) without actually knowing the impact, and then are surprised (or even "blame" qutebrowser) if websites start to break.

Having a command to "turn on privacy" will only make that worse - if there was no trade-off to those settings, those would all be the defaults already. Unfortunately, there are trade-offs, and I find it important that people can make an informed decision rather than just switching things on without knowing what they're actually doing (nevermind the fact that e.g. the things in the Arch Wiki most likely make you more fingerprintable rather than less so, because suddenly you have a browser which acts like Chromium but claims to be Firefox - that's going to be about as unique as it gets!).

**Comment by The-Compiler** [comment]
## Cleaning up handling of Chromium / Qt flags

This is a bit of a mess at the moment. There are three ways to pass arguments to Qt and/or the underlying Chromium:

- `--qt-flag` which takes a single flag (e.g. `--qt-flag no-sandbox` -> `--no-sandbox` for Chromium)
- `--qt-arg` which takes a name and a value (e.g. `--qt-arg name foobar` -> `--name foobar` for Qt, which sets `WM_NAME`)
- `qt.args` in the config, which actually corresponds to `--qt-flag`

There are various issues with those:

- It would be nice to separate flags passed to Qt/Chromium, and use `QTWEBENGINE_CHROMIUM_FLAGS` to do so
- There's currently no way to do the equivalent of `--qt-arg` in the config

Unfortunately, many ways of improving the situation mean that there would be a silent breakage (i.e. flags just stop being passed) which is kind of bad. The best thing I can think of right now:

- Rename `qt.args` to something like `qt.flags` (the main point is consistency with `--qt-flag` as well as deliberately introducing a backwards-incompatible change so people notice something changed)
- Add a `qt.chromium.flags` or so (could as well move `qt.low_end_device_mode`, `qt.force_software_rendering` and `qt.process_model` under `qt.chromium`, I guess)
- Not sure how to best handle the `--qt-arg` equivalent. Perhaps require the dashes (`--`) in the config rather than adding them - this would allow e.g. setting `qt.flags` to `["--name", "foo"]` to get the desired behavior. However, it'd mean `qt.flags` is slightly different from `--qt-flag` again. The only other thing I can think of is detecting `=` in values, so that `qt.flags` can be `["name=foo"]` to pass `--name foo` to Qt. It seems like a bit much magic, but maybe it's the better alternative. I *think* it's only needed for Qt flags, as Chromium flags must already be passed like `blinkSettings=...`.
- For the automatic config migration, migrate `qt.args` to `qt.chromium.flags` - it's somewhat unlikely people actually have flags for Qt in there. That's still some silent breakage, but much better than treating them as Qt flags. Maybe if we find out about certain Qt arguments people use, we could hardcode those to be migrated to `qt.flags` instead.

It's kind of a tricky situation - if anyone has ideas on how to handle this better, I'm all ears!

**Reply by rcorre** [reply]
> The only other thing I can think of is detecting = in values

Its a bit of a bummer if Qt doesn't understand that format natively, but I still think this is the best option. Having separate flags/configs for options that do/don't take arguments is confusing, and this would let us use a single CLI flag too, e.g. 

```
qutebrowser --qt-flag --no-sandbox --qt-flag --name=foobar
```

Then config would just have `qt.flag` as well, with `--` required as you said.

**Reply by The-Compiler** [reply]
I don't think we can parse something like `--qt-flag --no-sandbox --qt-flag --name=foobar` reasonably with Python's argparse - also, if we do require `--` (in `qt.flags`), we don't necessarily need the `=` because you can set it to `["--name", "foo"]`. If we add the `=` syntax, we don't need to require `--` because you can set it to `["name=foo"]`.

**Reply by The-Compiler** [reply]
I opened https://github.com/qutebrowser/qutebrowser/issues/6065 for this.

**Comment by arza-zara** [comment]
## Increase `messages.timeout` to 5000

Especially after https://github.com/qutebrowser/qutebrowser/pull/6013. Also consistent with `input.partial_timeout`.

**Reply by gavtroy** [reply]
Maybe an extra second, but for the messages typically shown I'd find 5 seconds very annoying. input.partial_timeout isn't really comparable.

**Reply by The-Compiler** [reply]
Raised to 3000 for now. Will reevaluate after #6013 (or an alternative) is in.

**Comment by arza-zara** [comment]
## Decrease `tabs.width`
Maybe 15% or 200 pixels.

**Comment by arza-zara** [comment]
## Blank or internal page in `url.default_page` and `url.start_pages`
Automatically opening a remote web page annoys me every time I open a clean instance.

**Reply by palb91** [reply]
For that I'm not sure that's it's a great idea to have it by default. I personnally set both to `about:blank` but I think that newcomers would prefer to have a sweet duck that kindly ask them what to do on startup.
I really think that "disabling them" belongs to the customization part.

**Reply by The-Compiler** [reply]
I'll need to think about it for a bit. I suppose we could still show a `qute://start` page with a search field going to DuckDuckGo - but I'm not sure if we could e.g. add their autocompletion. I've tried to reach out to them for a partnership but never got a reply...

**Reply by gerion0** [reply]
For me, a blank page is [also](https://github.com/qutebrowser/qutebrowser/discussions/5999#discussioncomment-257746) a privacy feature that should be enabled in a "privacy focused" qutebrowser configuration but not as default.

**Reply by The-Compiler** [reply]
I opened https://github.com/qutebrowser/qutebrowser/issues/6066 for this.

**Comment by The-Compiler** [comment]
## Changing the default search engine

Search engines are the "traditional" way browsers make money, and something qutebrowser so far hasn't done - I've tried reaching out to DuckDuckGo for a partnership in 2016 (where they claimed qutebrowser was too small to be interesting for them) and again in July 2020 (with no answer).

However, Ecosia has reached out to me in the past, see #1682 - over there, people seemed to be in favor of the change, but I'm still kinda hesitant, especially because we'd lose duckduckgo bangs. Perhaps with a custom start page (see above) we could still hand those to DuckDuckGo, though.

Perhaps another more common alternative is interested in a partnership, such as [Startpage](https://www.startpage.com/) (see https://github.com/privacytools/privacytools.io/issues/1409 though) or [Qwant](https://lite.qwant.com/) - or perhaps [Mojeek](https://www.mojeek.com/) or [swisscows](https://swisscows.com/), though they hardly qualify as "more common" I guess.

**Reply by formerCIAjanitor** [reply]
FWIW, I was using DuckDuckGo only for the bangs...until I found I could make my own, e.g.:

`c.url.searchengines = { "w": 'https://en.wikipedia.org/wiki/{}', "gv": 'https://www.google.com/search?tbm=vid&q={}'}`

Less keystrokes and more customization, a much better solution IMHO. 

I still can use DuckDuckGo bangs though, thanks to `"d": 'https://duckduckgo.com/{}'`, which means I just need to add "d " before the bang. 

If it was my decision, I would not hesitate to pick the search engines the most beneficial to the project and put the above in the wiki (it is probably already there, if not make it easier to find?) and ask users to share their own "bangs" in a wiki page as the url strings are sometimes not obvious.
I'd share mine ;)

**Reply by markstos** [reply]
If this helps you financially, go for it. It just takes a few seconds for users to change the default search engine if they care to.

**Reply by The-Compiler** [reply]
~~I've opened https://github.com/qutebrowser/qutebrowser/issues/6067 for this.~~

I think this is tracked via #1682 already.

**Comment by The-Compiler** [comment]
FYI: I've hidden some sub-threads where I felt like a conclusion has been reached, and summarized the outcomes in #6022. This should make this a bit simpler to scroll through, and avoids hiding the later discussion points below a lot of text. Please keep the suggestions coming, and feel free to let me know if you feel like there's something new to be said about a thread I've been hidden (e.g. below this reply, as for some reason hiding a thread seems to remove the ability to reply).

**Comment by gavtroy** [comment]
## Set `input.partial_timeout` to `0`.

There's often a lot of text to read in 5 seconds. Furthermore the user is actively engaging in input and can easily hit Esc if they want to. It also avoids a race with the user entering more keystrokes.

**Reply by The-Compiler** [reply]
FWIW that setting was added (in e696898c4a35c4a9b487969d765313d132f6e848) before the keyhint window existed.  I think it was because some people (myself included) sometimes accidentally press some key without noticing, which then can be confusing when trying to start a new keychain at a later point.

With the keyhint window, I guess "pressing a key without noticing" isn't a thing that happens at all anymore, so I guess that setting can be set to 0 indeed. I'm now actually thinking about removing it entirely, but I'm sure someone would complain for some reason. :laughing:

**Comment by bodograumann** [comment]
## tab-move left / right

Currently there is `gr` and `gl` to move tabs. This is not very ideomatic.
What about adding `gK` and `gJ` as default bindings?

**Reply by The-Compiler** [reply]
I'm kinda used to `gr` and `gl`, but I agree it's weird (especially with vertical tabs). `gK` and `gJ` makes much more sense, so I'm sold!

**Comment by The-Compiler** [comment]
## Change handling of DEFAULT searchengine

The `DEFAULT` key in `url.searchengines` always felt like a bit of an oddball to me. I think it'd be nicer to have something like `url.searchengines.names` (or so, can't think of a good name) which is like the current `url.searchengines` dict, and then have `url.searchengines.default` which points to one of those names. We could even have a `url.searchengines.default_bang` or so then, which would be used when the search term contains a `!bang` so that it'd be possible to easily use a different engine but still use DuckDuckGo for its bangs feature.

Might be harder to validate the setting properly though... Not sure if we can validate it at all (as the settings can be set in any order).

**Reply by The-Compiler** [reply]
I've opened https://github.com/qutebrowser/qutebrowser/issues/6067 for this.

**Comment by rien333** [comment]
# Change the name of the "Meta" modifier to "Super"

Given that we're offering ideas for changing names, I would like to point a fairly qutebrowser-specific naming-convention that has always bothered me. The modifier that's currently designated by "Meta" is, within the Linux world, colloquially referred to as the "Super" key (or "command" in the macOS community).  Indeed, emacs, `xev` (X11), kitty, sxkhd, an so on, internally refer to this modifier as "Super".  I even have a keyboard that labels this key "Super".

I can't really figure out where the label "Meta" arose from.  [Historically](https://en.wikipedia.org/wiki/Meta_key), the meta and super keys have been different things. In fact, within the emacs (and Vim?) community meta is often used to refer to the <kbd>Alt</kbd> or <kbd>Esc</kbd> keys. Espcially this latter fact trips me up immensenly, as qutebrowser and emacs are two of my most often used programs.

Some fun historical and modern-day pictures of keyboard with super keys (note the seperate meta key in the first picture):

![330px-Space-cadet_keyboard_-_bottom-left_modifier_crop](https://user-images.githubusercontent.com/10417027/105239549-237eaf80-5b6d-11eb-93ef-91807ca08964.jpg)

![t](https://user-images.githubusercontent.com/10417027/105240382-588b0200-5b6d-11eb-873e-3f25f3aef6b7.jpg)

**Reply by rien333** [reply]
I think the KDE docs are the only ones that refer to this key as "Meta" (Ubuntu and Gnome don't, see e.g. https://help.gnome.org/users/gnome-help/stable/keyboard-key-super.html.en)

**Reply by rien333** [reply]
Ah, it's a thing internal to Qt: https://bugreports.qt.io/browse/QTBUG-61301

```c
 // translate Super/Hyper keys to Meta if we're using them as the MetaModifier
        if (rmod_masks.meta == rmod_masks.super && (qtKey == Qt::Key_Super_L
                                                 || qtKey == Qt::Key_Super_R)) {
            qtKey = Qt::Key_Meta;
// [rien333] do the same thing for Hyper
```

I kinda feel like this translation of Super to Meta is a case in point. Still, I also understand if implementation-wise, it's easier to refer to this key as "Meta".

**Reply by bodograumann** [reply]
Yes for vim, meta also means usually the ALT key. From `:help meta`:

> <M-...>		alt-key or meta-key		*meta* *alt* *<M-*
> <A-...>		same as <M-...>			*<A-*

For xbd there is (was?) `alt_is_meta`:
> alt_is_meta
>     Whenever some combination is defined as an ASCII symbol, and there is a corresponding Alt keymap, define by default the corresponding Alt combination as Meta_value. 
I don’t see it in current keymaps anymore though.

The [http://zsh.sourceforge.net/Guide/zshguide04.html](zsh guide) also refers to a “meta key”, but says

> In X Windows, one of the keys on your keyboard, possibly ALT, may be designated a `meta' key which has a special effect similar to the control key.

This can be seen in xkb, where there are `Alt_L`, `Meta_L`, `Alt_R` and `Meta_R` symbols. Often they are mapped the same keys, but it is possible to change that with xkb options. Some even have seperate Alt and Meta keys by default. From  /usr/share/X11/xkb/symbols/fr:

```
// Unicode French derivative
// Loose refactoring of the historic Linux French keyboard layout
//
// Copyright © 2006-2008 Nicolas Mailhot <nicolas.mailhot @ laposte.net>
//
// Credits (fr-latin1, fr-latin0, fr-latin9)
//   © 199x-1996 René Cougnenc ✝
//   © 1997-2002 Guylhem Aznar <clavier @ externe.net>
//   © 2003-2006 Nicolas Mailhot <nicolas.mailhot @ laposte.net>
//
// ┌─────┬─────┬─────┬─────┬─────┬─────┬─────┬─────┬─────┬─────┬─────┬─────┬─────┲━━━━━━━━━┓
// │ ³ ¸ │ 1 ̨  │ 2 É │ 3 ˘ │ 4 — │ 5 – │ 6 ‑ │ 7 È │ 8 ™ │ 9 Ç │ 0 À │ ° ≠ │ + ± ┃ ⌫ Retour┃
// │ ² ¹ │ & ˇ │ é ~ │ " # │ ' { │ ( [ │ - | │ è ` │ _ \ │ ç ^ │ à @ │ ) ] │ = } ┃  arrière┃
// ┢━━━━━┷━┱───┴─┬───┴─┬───┴─┬───┴─┬───┴─┬───┴─┬───┴─┬───┴─┬───┴─┬───┴─┬───┴─┬───┺━┳━━━━━━━┫
// ┃       ┃ A Æ │ Z Â │ E ¢ │ R Ê │ T Þ │ Y Ÿ │ U Û │ I Î │ O Œ │ P Ô │ ¨ ˚ │ £ Ø ┃Entrée ┃
// ┃Tab ↹  ┃ a æ │ z â │ e € │ r ê │ t þ │ y ÿ │ u û │ i î │ o œ │ p ô │ ^ ~ │ $ ø ┃   ⏎   ┃
// ┣━━━━━━━┻┱────┴┬────┴┬────┴┬────┴┬────┴┬────┴┬────┴┬────┴┬────┴┬────┴┬────┴┬────┺┓      ┃
// ┃        ┃ Q Ä │ S „ │ D Ë │ F ‚ │ G ¥ │ H Ð │ J Ü │ K Ï │ L Ŀ │ M Ö │ % Ù │ µ ̄  ┃      ┃
// ┃Maj ⇬   ┃ q ä │ s ß │ d ë │ f ‘ │ g ’ │ h ð │ j ü │ k ï │ l ŀ │ m ö │ ù ' │ * ` ┃      ┃
// ┣━━━━━━━┳┹────┬┴────┬┴────┬┴────┬┴────┬┴────┬┴────┬┴────┬┴────┬┴────┬┴────┲┷━━━━━┻━━━━━━┫
// ┃       ┃ > ≥ │ W “ │ X ” │ C ® │ V ← │ B ↑ │ N → │ ? … │ . . │ / ∕ │ § − ┃             ┃
// ┃Shift ⇧┃ < ≤ │ w « │ x » │ c © │ v ⍽ │ b ↓ │ n ¬ │ , ¿ │ ; × │ : ÷ │ ! ¡ ┃Shift ⇧      ┃
// ┣━━━━━━━╋━━━━━┷━┳━━━┷━━━┱─┴─────┴─────┴─────┴─────┴─────┴───┲━┷━━━━━╈━━━━━┻━┳━━━━━━━┳━━━┛
// ┃       ┃       ┃       ┃ ␣         Espace fine insécable ⍽ ┃       ┃       ┃       ┃
// ┃Ctrl   ┃Meta   ┃Alt    ┃ ␣ Espace       Espace insécable ⍽ ┃AltGr ⇮┃Menu   ┃Ctrl   ┃
// ┗━━━━━━━┻━━━━━━━┻━━━━━━━┹───────────────────────────────────┺━━━━━━━┻━━━━━━━┻━━━━━━━┛
partial alphanumeric_keys
xkb_symbols "oss" {
```

**Reply by The-Compiler** [reply]
qutebrowser already understood "windows", "mod4", "command" and "cmd" as aliases, but for some reason "super" was missing - since it's a trivial change and shouldn't affect people using the other names, I've added "super" in 41afe0febe437e73df4edb3548036eedc0ee70c4.

**Reply by rien333** [reply]
yaayy, feel free to hide this now!

**Comment by The-Compiler** [comment, marked as answer]
I think everything discussed so far should either be implemented or in a separate issue now. Thanks everyone for all the suggestions!

**Comment by godalming123** [comment]
I think that there should be `ge` keybinding to go to the end of a web page as well as the `G` keybinding since `ge` is the keybindings that [helix](https://github.com/helix-editor/helix) uses to go to the end of a file, and I don't see any disadvantage of having 2 keybindings to do this since `ge` currently doesn't do anything.
